libpromises/locks.c

```diff
@@ -468,14 +468,67 @@ static void PromiseTypeString(char *dst, size_t dst_size, const Promise *pp)
 
 #ifdef __MINGW32__
 
-static bool KillLockHolder(ARG_UNUSED const char *lock)
+// Windows only functions for killing a process
+// NOTE: GetLastError(), FormatMessage(), OpenProcess() and TerminateProcess() are Windows API functions
+
+static void WindowsLogError()
 {
-    Log(LOG_LEVEL_VERBOSE,
-          "Process is not running - ignoring lock (Windows does not support graceful processes termination)");
-    return true;
+    int error_code = GetLastError();
+    Log(LOG_LEVEL_ERR, "Windows API error code: %i", error_code);
+
+    char * error_message = NULL;
+
+    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM|FORMAT_MESSAGE_ALLOCATE_BUFFER
+                  |FORMAT_MESSAGE_IGNORE_INSERTS,
+                  NULL, error_code,
+                  MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                  &error_message, 0, NULL);
```

Comment:
Why do you think that? `GetLastError()` returns an error code, not a readable string. This is the purpose of `FormatMessage()`.

> The FormatMessage function can be used to obtain error message strings for the system error codes returned by GetLastError. For an example, see Retrieving the Last-Error Code.

My code is very similar to the MSDN example: https://msdn.microsoft.com/en-us/library/windows/desktop/ms679351(v=vs.85).aspx